edx-solutions/xblock-drag-and-drop-v2
# Problem mode field and edit capabilities

Reincarnation of #75 

**Description:** This PR adds a mode field to DnDv2 with to possible values: "standard" and "assessment" + studio editing capabilities.

This is the first of a series of PRs that will be implementing the new "[Assessment mode](https://docs.google.com/document/d/1rFliZo1vlohJ7cde7KfP5bgEk0QqcvT1VNnUUa9_vjE/edit)". This story is only adding the editor UI, and does not yet implement functionality changes, which are coming in subsequent stories.

**Dependencies:** Based on https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/74
**JIRA:** [SOL-1933](https://openedx.atlassian.net/browse/SOL-1933)

**Sandbox LMS URL**: http://dndv2-sandbox2.opencraft.hosting/
**Sandbox Studio URL**: http://studio-dndv2-sandbox2.opencraft.hosting/

**Screenshot:**
![image](https://cloud.githubusercontent.com/assets/3330048/16801438/55014b20-4903-11e6-88f3-0da53a700295.png)


**Testing instructions:**
1. Install XBlock into LMS/Studio
2. Enable in Advanced settings
3. Create a course, section, subsection and unit
4. Add DnDv2 block to a unit
5. Click edit. *Expected result*: Editor popup opens; "Problem mode" select with two values "standard" and "assessment" are present, with "standard" selected
6. Change "Problem mode", save DnDv2 (note the edit dialog have three steps)
7. Click "Edit" again. *Expected result*: Editor popup opens, "Problem mode" is present and previously saved value is selected.

Reviewers:

* [x] @cahrens
